mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -72,97 +77,107 @@ public DefaultEntityFactory(ProcessorContext context) {
+          continue;
+        }
+        TypeMirror typeMirror = getMethod.getReturnType();
+        if (typeMirror.getKind() == TypeKind.VOID) {
+          continue;
+        }
+        String propertyName = Introspector.decapitalize(getMethodName.substring(3));
+        // skip properties we've already encountered.
+        if (encounteredPropertyNames.contains(propertyName)) {
+          continue;
+        }
 
-      int partitionKeyIndex = getPartitionKeyIndex(propertyAnnotations);
-      int clusteringColumnIndex = getClusteringColumnIndex(propertyAnnotations);
-      Optional<String> customCqlName = getCustomCqlName(propertyAnnotations);
-      Optional<String> computedFormula = getComputedFormula(propertyAnnotations, getMethod, field);
-
-      PropertyType propertyType = PropertyType.parse(typeMirror, context);
-      PropertyDefinition property =
-          new DefaultPropertyDefinition(
-              propertyName,
-              customCqlName,
-              computedFormula,
-              getMethodName,
-              setMethodName,
-              propertyType,
-              cqlNameGenerator);
-
-      if (partitionKeyIndex >= 0) {
-        PropertyDefinition previous = partitionKey.putIfAbsent(partitionKeyIndex, property);
-        if (previous != null) {
-          context
-              .getMessager()
-              .error(
-                  getMethod,
-                  "Duplicate partition key index: if multiple properties are annotated "
-                      + "with @%s, the annotation must be parameterized with an integer "
-                      + "indicating the position. Found duplicate index %d for %s and %s.",
-                  PartitionKey.class.getSimpleName(),
-                  partitionKeyIndex,
-                  previous.getGetterName(),
-                  property.getGetterName());
+        String setMethodName = getMethodName.replaceFirst("get", "set");
+        ExecutableElement setMethod = findSetMethod(typeElement, setMethodName, typeMirror);
+        if (setMethod == null) {
+          continue; // must have both
         }
-      } else if (clusteringColumnIndex >= 0) {
-        PropertyDefinition previous =
-            clusteringColumns.putIfAbsent(clusteringColumnIndex, property);
-        if (previous != null) {
-          context
-              .getMessager()
-              .error(
-                  getMethod,
-                  "Duplicate clustering column index: if multiple properties are annotated "
-                      + "with @%s, the annotation must be parameterized with an integer "
-                      + "indicating the position. Found duplicate index %d for %s and %s.",
-                  ClusteringColumn.class.getSimpleName(),
-                  clusteringColumnIndex,
-                  previous.getGetterName(),
-                  property.getGetterName());
+        VariableElement field = findField(typeHierarchy, propertyName, typeMirror);
```

Comment:
Might need some logic here to bound the highest type parsed for the field to where the getMethod was defined...not sure if that's worth it or not though.

Reply:
It seems unlikely that a field with the same name would exist at a higher level _and_ be annotated,  I think we can safely ignore that case.